src/main.rs

```diff
@@ -61,6 +63,33 @@ fn load_filetypes_database(matches: &ArgMatches, user_config_path: &PathBuf) ->
     }
 }
 
+fn available_theme_names(user_config_path: &PathBuf) -> Result<Vec<String>> {
+    let theme_path_user = user_config_path.clone().join("themes");
+    let theme_path_system = PathBuf::from(THEME_PATH_SYSTEM);
+    let theme_path = util::get_first_existing_path(&[&theme_path_user, &theme_path_system]);
+
+    let mut theme_names = match theme_path {
+        Some(path) => {
+            let dir = fs::read_dir(path).map_err(VividError::IoError)?;
+            let mut available_themes = vec![];
+            for theme_file in dir {
+                let theme_name = theme_file
+                    .map_err(VividError::IoError)?
+                    .file_name()
+                    .into_string()
+                    .map_err(|n| VividError::InvalidFileName(n.into_string().unwrap()))?;
```

Comment:
If `into_string` has failed in the line above this one, it will also fail here inside `map_err`. Which, in turn, will cause `.unwrap()` to panic.

You can try this by running
```
touch ~/.config/vivid/themes/$(printf 'whatever\xFE')
```
before trying to list the themes.

What you could do instead (either in the error handling, or directly for listing the themes) would be to use `.as_os_str().to_string_lossy()` which replaces invalid UTF-8 characters with a `�` symbol.

Reply:
Ah yes, thanks for the detailed explanation, it did seem like bad practice to `unwrap` there, makes sense now.